n3fit/src/n3fit/scripts/evolven3fit_new.py

```diff
@@ -0,0 +1,376 @@
+"""
+This module contains the CLI for evolven3fit_new
+"""
+import logging
+import pathlib
+from argparse import ArgumentParser
+import sys
+import numpy as np
+
+from evolven3fit_new import eko_utils, evolve, utils
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    parser = ArgumentParser(
+        description="evolven3fit_new - a script with tools to evolve PDF fits"
+    )
+    subparsers = parser.add_subparsers(title="actions", dest="actions")
+    evolvefit_parser = subparsers.add_parser(
+        "evolve",
+        help="Evolves the fitted PDFs. The q_grid starts at the Q0 given by the theory but the last point is q_fin and its number of points can be specified by q_points. If a path is given for the dump option, the eko will be dumped in that path after the computation. If a path is given for the load option, the eko to be used for the evolution will be loaded from that path. The two options are mutually exclusive.",
+    )
+    eko_parser = subparsers.add_parser(
+        "produce_eko",
+        help="Produce the eko for the specified theory. The q_grid starts at the Q0 given by the theory but the last point is q_fin and its number of points can be specified by q_points. The x_grid starts at x_grid_ini and ends at 1.0 and contains the provided number of points. The eko will be dumped in the provided path.",
+    )
+    evolvefit_parser.add_argument(
+        "configuration_folder",
+        help="Path to the folder containing the (pre-DGLAP) fit result",
+    )
+    parser.add_argument(
+        "-q",
+        "--q_fin",
+        type=float,
+        default=None,
+        help="Final q-value of the evolution",
+    )
+    parser.add_argument(
+        "-p",
+        "--q_points",
+        type=int,
+        default=None,
+        help="Number of q points for the evolution",
+    )
+    parser.add_argument(
+        "-n",
+        "--n_cores",
+        type=int,
+        default=1,
+        help="Number of cores to be used",
+    )
+    evolvefit_parser.add_argument(
+        "-l",
+        "--load",
+        type=pathlib.Path,
+        default=None,
+        help="Path of the EKO to be loaded",
+    )
+    evolvefit_parser.add_argument(
+        "-d",
+        "--dump",
+        type=pathlib.Path,
+        default=None,
+        help="Path where the EKO is dumped (optional)",
```

Comment:
You are writing `(optional)` inconsistently in various arguments.

However, all named arguments (starting with dashes) should be optional, otherwise they should be positional. You can also specify [`required = True`](https://docs.python.org/3/library/argparse.html?highlight=argparse#required) for a named (and if it is required you should really put it, instead of calling the others `(optional)` in the help strings), but it is not an intuitive interface, so try to avoid.

Reply:
> Note
>
> Required options are generally considered bad form because users expect options to be optional, and thus they should be avoided when possible.

Reply:
Actually all the named arguments are really optional as they should, not only `--dump`. So I can just remove `(optional)` from its `help` and that's it.